src/modules/ens/sagas.ts

```diff
@@ -0,0 +1,129 @@
+import { Eth } from 'web3x-es/eth'
+import { call, put, takeEvery, select } from 'redux-saga/effects'
+import {
+  GET_ENS_REQUEST, 
+  GetENSRequestAction, 
+  getENSSuccess, 
+  getENSFailure, 
+  SET_ENS_REQUEST,
+  SetENSRequestAction, 
+  setENSSuccess, 
+  setENSFailure
+} from './actions'
+import { ENS } from 'contracts/ENS'
+import { ENSResolver } from 'contracts/ENSResolver'
+import { Address } from 'web3x-es/address'
+import { getAddress } from 'decentraland-dapps/dist/modules/wallet/selectors'
+import { 
+  ENS_ADDRESS, 
+  ENS_RESOLVER_ADDRESS 
+} from 'modules/common/contracts'
+import { ipfs } from 'lib/api/ipfs'
+import { namehash } from "@ethersproject/hash";
+import * as contentHash from 'content-hash'
+import { ENS_EMPTY_CONTENT, ENS_EMPTY_RESOLVER } from './constants'
+
+export function* ensSaga() {
+  yield takeEvery(GET_ENS_REQUEST, handleGetENSRequest)
+  yield takeEvery(SET_ENS_REQUEST, handleSetENSRequest)
+}
+
+function* handleSetENSRequest(action: SetENSRequestAction) {
+  const { land, ens } = action.payload
+  try {
+    const [eth, from] = yield getEth()
+    const nodehash = namehash(ens)
+    const ensContract = new ENS(eth, Address.fromString(ENS_ADDRESS))
+    const resolverContract = new ENSResolver(eth, Address.fromString(ENS_RESOLVER_ADDRESS))
+
+    const ipfsHash = yield call(() => ipfs.uploadRedirectionFile(land))
+    const hash = contentHash.fromIpfs(ipfsHash)
+    yield call(() => 
+      resolverContract.methods
+        .setContenthash(nodehash, `0x${hash}`)
+        .send({from})
+        .getTxHash()
+    )
+    yield call(() => 
```

Comment:
this is always sending the `setResolver` transaction isn't it? Shouldn't we do that only if the resolver is not set?

Reply:
If it have another resolver and the user want to change it to our resolver, then we need to change it. Or am I missing something?

Reply:
The resolver is the _public_ one. Is not _our_. We can check if the user is using the public resolver, if not. We can send the tx to change it.